### PR TITLE
adding tenant flow for update tenant

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

**Comment by johannnallathamby** [comment]
Reviewed. Can merge.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 1]
```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
...
+            PrivilegedCarbonContext.startTenantFlow();
+
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            carbonContext.setTenantDomain(tenantDomain);
+            carbonContext.setTenantId(tenantId);
+
+            TenantMgtUtil.addClaimsToUserStoreManager(tenant);
+
+            // filling the email value
+            if (tenantInfoBean.getEmail() != null && !tenantInfoBean.getEmail().equals("")) {
```
Comment (pulasthi7) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Use StringUtils.isNotBlank()

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 2]
```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
...
+                // validate the email
+                try {
+                    CommonUtil.validateEmail(tenantInfoBean.getEmail());
+                } catch (Exception e) {
+                    String msg = "Invalid email is provided.";
+                    log.error(msg, e);
+                    throw new Exception(msg, e);
```
Comment (pulasthi7) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: 'e' is both logged and thrown. is there any specific reason?

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 3]
```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
...
                 log.error(msg, e);
                 throw new Exception(msg, e);
             }
-        } else {
-            //Password should be empty since no password update done
-            tenantInfoBean.setAdminPassword("");
+
+            boolean updatePassword = false;
+            if (tenantInfoBean.getAdminPassword() != null
+                    && !tenantInfoBean.getAdminPassword().equals("")) {
```
Comment (pulasthi7) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Use StringUtils.isNotBlank()

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java [hunk 4]
```diff
@@ -391,52 +391,63 @@ public void updateTenant(TenantInfoBean tenantInfoBean) throws Exception {
...
+                // password.
+                try {
+                    userStoreManager.updateCredentialByAdmin(tenantInfoBean.getAdmin(),
+                            tenantInfoBean.getAdminPassword());
+                } catch (UserStoreException e) {
+                    String msg = "Error in changing the tenant admin password, tenant domain: " +
+                            tenantInfoBean.getTenantDomain() + ". " + e.getMessage() + " for: " +
+                            tenantInfoBean.getAdmin();
+                    log.error(msg, e);
+                    throw new Exception(msg, e);
```
Comment (pulasthi7) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: 'e' is both logged and thrown. is there any specific reason?

Comment (madurangasiriwardena) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: This code segment was already there. Due to formatting it is shown as a new addition.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 2 above)
Comment (madurangasiriwardena) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: This code segment was already there. Due to formatting it is shown as a new addition.

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 1 above)
Comment (madurangasiriwardena) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Fixed with https://github.com/wso2/carbon-multitenancy/pull/75

File: components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java (hunk 3 above)
Comment (madurangasiriwardena) [inline components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java]: Fixed with https://github.com/wso2/carbon-multitenancy/pull/75

**Pull request merged**